### PR TITLE
config: add validation for capacity of prepare plan cache

### PR DESCRIPTION
**Pull request by eurekaka**
### What problem does this PR solve? 

```
$ cat config.toml
[prepared-plan-cache]
enabled = true
capacity = 0

$ ./bin/tidb-server -config config.toml -config-check
config check successful

$ ./bin/tidb-server -config config.toml -- panic is reported
```

### What is changed and how it works?

Add validation for capacity config of prepare plan cache.

### Check List 

Tests 

 - Manual test: no panic after applying the fix for the above steps.

Code changes

N/A

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Write release note for bug-fix or new feature.
```
Report error for invalid capacity configuration of prepare plan cache
```

### CODE REVIEW

File: util/kvcache/simple_lru.go [hunk 1]
```diff
@@ -48,8 +48,8 @@ type SimpleLRUCache struct {
 // NewSimpleLRUCache creates a SimpleLRUCache object, whose capacity is "capacity".
 // NOTE: "capacity" should be a positive value.
```
Comment (Deardrops) [inline util/kvcache/simple_lru.go]: Rewriting this comment also?

File: util/kvcache/simple_lru.go [hunk 2]
```diff
@@ -48,8 +48,8 @@ type SimpleLRUCache struct {
...
 func NewSimpleLRUCache(capacity uint, guard float64, quota uint64) *SimpleLRUCache {
-	if capacity == 0 {
-		panic("capacity of LRU Cache should be positive.")
+	if capacity < 1 {
```
Comment (wjhuang2016) [inline util/kvcache/simple_lru.go]: `uint` imply >=0, is it necessary to update this code?

Comment (eurekaka) [reply util/kvcache/simple_lru.go]: Just to update the comment to be consistent with configuration check.

**Pull request merged**